src/anyio/_backends/_asyncio.py

```diff
@@ -190,11 +189,15 @@ def __init__(self, deadline: float = math.inf, shield: bool = False):
         self._host_task: Optional[asyncio.Task] = None
         self._timeout_expired = False
 
+    def _timeout(self):
+        self._timeout_expired = True
+        self._cancel_called = True
+        self._cancel()
+
     def __enter__(self):
         async def timeout():
             await asyncio.sleep(self._deadline - get_running_loop().time())
-            self._timeout_expired = True
-            self.cancel()
+            self._timeout()
```

Comment:
Do we need this coroutine function here any more now that `cancel()` is a regular method?

Reply:
`self._timeout()` is a regular function

Reply:
Yeah, so what? We'll just use `loop.call_later()`.

Reply:
seems to fail the tests: https://github.com/agronholm/anyio/pull/207/checks?check_run_id=1962019106#step:6:40